builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +81,44 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
+	// Get the role name
+	roleNameRaw, ok := req.Secret.InternalData["role"]
+	roleName, ok := roleNameRaw.(string)
+
+	// init role entry struct
+	var role = &roleEntry{}
```

Comment:
Rather than go through this rigamarole, simply declare a `revokeSQL` string that is equivalent to the default (`revokeSQL := defaultRevokeSQL`).. Then if the `if` condition passes, you can fetch the role, and if `role.RevokeSQL` isn't empty you can assign `revokeSQL` to it, all within the `if` block.